### PR TITLE
Extend Get MachineDeployment upgrades for EKS clusters

### PR DESCRIPTION
**Pull request by imharshita**
Signed-off-by: Harshita sharma <harshita.sharma6174@gmail.com>

**What does this PR do / Why do we need it**: Extend Get MachineDeployment upgrades for EKS clusters

**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: pkg/provider/cloud/eks/provider.go [hunk 1]
```diff
@@ -198,3 +200,56 @@ func convertEKSStatus(status string) apiv2.ExternalClusterState {
...
+	}
+
+	nodeGroupOutput, err := client.EKS.DescribeNodegroup(nodeGroupInput)
+	if err != nil {
+		return nil, err
+	}
+	nodeGroup := nodeGroupOutput.Nodegroup
+
+	if nodeGroup.Version == nil {
+		return nil, fmt.Errorf("unable to get EKS cluster %s nodegroup %s  version", clusterName, machineDeployment)
```
Comment (lsviben) [inline pkg/provider/cloud/eks/provider.go]: very nitty:
```suggestion
		return nil, fmt.Errorf("unable to get EKS cluster %s nodegroup %s version", clusterName, machineDeployment)
```

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**